README.md
V 0.4
- zmieniono położenie pokazyania wyników
- sprawdza wszystkie elementy na stronie a nie tylko 5
- odbugowano gdy jeden z elementów został sprzedany w momencie wejścia na stronę
- pozbyto się niektórych fragmentów jquery

V 0.3.1
- poprawiono sprawdzanie cen powyżej 999zł
- poparwiono błąd powodujący wysypanie się skryptu przy ilości ofert sprzedaży mniejszej niż 5 

- V 0.3 
- poprawa błędu  wyświetlającego undefined 
- zmiana zmiennych na angielski
- poprawa sprawdzania ceny zlecenia ( teraz wykonuje się raz na początku a nie za każdym sprawdzeniem cen wystawionych przedmiotów)
- kolorowanie profitu w zależności od zysków/strat

V 0.2
- wyświetlanie zysków w body

V 0.1
- sprawdzenie ceny zlecenia
- sprawdzenie 5 pierwszych ofert
- obliczanie zysków dla 5 ofert
- wyświeltenie zysków jako alerty


CHANGELOG
V 0.1
- sprawdzenie ceny zlecenia
- sprawdzenie 5 pierwszych ofert
- obliczanie zysków dla 5 ofert
- wyświeltenie zysków jako alerty

V 0.2
- wyświetlanie zysków w body

V 0.3 
- poprawa błędu  wyświetlającego undefined 
- zmiana zmiennych na angielski
- poprawa sprawdzania ceny zlecenia ( teraz wykonuje się raz na początku a nie za każdym sprawdzeniem cen wystawionych przedmiotów)
- kolorowanie profitu w zależności od zysków/strat

V 0.3.1
- poprawiono sprawdzanie cen powyżej 999zł
- poparwiono błąd powodujący wysypanie się skryptu przy ilości ofert sprzedaży mniejszej niż 5 

V 0.4
- zmieniono położenie pokazyania wyników
- sprawdza wszystkie elementy na stronie a nie tylko 5
- odbugowano gdy jeden z elementów został sprzedany w momencie wejścia na stronę
- pozbyto się niektórych fragmentów jquery


TradeHelper.js
// ==UserScript==
// @require      https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js
// @name         TH 
// @version      0.1
// @author       You
// @match        https://steamcommunity.com/market/listings/730/*
// @grant        none
// ==/UserScript==
var link = "https://steamcommunity.com/market/listings/730/";
$(document).ready(function() {
    if (location.href.substring(0, 47) == link) {
        setTimeout(find, 1000);
    }
})

var find = function(){
    var i=0;
    var text;
    for(i;i<5;i++){
        var zlecen = $('#market_commodity_buyrequests').text();
        zlecen = zlecen.slice(25, 200);
        zlecen = zlecen.match(/[0-9]*\,[0-9][0-9]/);
        zlecen = zlecen.toString().replace(",", ".");
        zlecen = parseFloat( zlecen );
        var cena = $('.market_listing_price_without_fee').eq(i).text();
        cena = cena.match(/[0-9]*\,[0-9][0-9]/);
        cena = cena.toString().replace(",", ".");
        cena = parseFloat( cena );
        var roznica = cena - zlecen;
        roznica = roznica.toString();
        roznica = roznica.match(/.[0-9]*\.[0-9][0-9]/);
        text = text + roznica + '\n';
    }
    alert(text);
};


TradeHelper v0.2.js
// ==UserScript==
// @require      https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js
// @name         TH xD
// @version      0.2
// @author       You
// @match        https://steamcommunity.com/market/listings/730/*
// @grant        none
// ==/UserScript==
let link = "https://steamcommunity.com/market/listings/730/";

$(document).ready(function() {
    if (location.href.substring(0, 47) == link) {
        setTimeout(checkPrices, 1000);
    }
})

let checkPrices = () => {
    let i=0;
    let text;
    for(i;i<5;i++){
        let zlecen = $('#market_commodity_buyrequests').text();
        zlecen = zlecen.slice(25, 200);
        zlecen = zlecen.match(/[0-9]*\,[0-9][0-9]/);
        zlecen = zlecen.toString().replace(",", ".");
        zlecen = parseFloat( zlecen );
        let cena = $('.market_listing_price_without_fee').eq(i).text();
        cena = cena.match(/[0-9]*\,[0-9][0-9]/);
        cena = cena.toString().replace(",", ".");
        cena = parseFloat( cena );
        let roznica = cena - zlecen;
        roznica = roznica.toString();
        roznica = roznica.match(/.[0-9]*\.[0-9][0-9]/);
        text = text + roznica + '\n';
    }
    show(text);
};

let show = (prices) => {
    $('<div/>',{
        text: prices,
        style: "background-color:red; color:white; font-size:20px; position: fixed; top:104px; text-align:center; width: 50%; left:50%; transform:translateX(-50%);"
    }).appendTo('body');
}

TradeHelper v0.3.js
// ==UserScript==
// @require      https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js
// @name         TH 0.3
// @version      0.3
// @author       You
// @match        https://steamcommunity.com/market/listings/730/*
// @grant        none
// ==/UserScript==

// CS:GO Market link
let link = "https://steamcommunity.com/market/listings/730/";

// Document ready function
$(document).ready(function() {
    if (location.href.substring(0, 47) == link) {
        setTimeout(checkRequest, 1000);
    }
})

// Check request value
let checkRequest = () => {
    let request = 0;
    request = $('#market_commodity_buyrequests').text();
    request = request.slice(25, 200);
    request = request.match(/[0-9]*\,[0-9][0-9]/);
    request = parseFloat(request[0].toString().replace(",", "."));
    checkPrices(request);
}

// Check Market Prices
let checkPrices = (request) => {
    let i=0;
    let profits = [];
    for(i;i<5;i++){

        let price = $('.market_listing_price_without_fee').eq(i).text();
        price = price.match(/[0-9]*\,[0-9][0-9]/);
        price = parseFloat(price.toString().replace(",", "."));
        let profit = price - request;
        profit = profit.toFixed(2);
        profits.push(parseFloat(profit));
    }
    show(profits);
};

// Show informations
let show = (profits) => {
    let i = 0;
    let color = 0;
    let text = "";

    for(i;i<5;i++){
        if(profits[i] > 0)
            color = "green";
        else
            color = "red";
        text = text + "<span style='color:"+color+"; margin:10px'>"+profits[i]+"</span>";
    }

    $('<div/>',{
        html: text,
        style: "background-color:black; color:white; font-size:20px; position: fixed; top:104px; text-align:center; width: 50%; left:50%; transform:translateX(-50%);"
    }).appendTo('body');
}

TradeHelper v0.3.1.js
// ==UserScript==
// @require      https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js
// @name         TH 0.3.1
// @version      0.3.1
// @author       You
// @match        https://steamcommunity.com/market/listings/730/*
// @grant        none
// ==/UserScript==

// CS:GO Market link
let link = "https://steamcommunity.com/market/listings/730/";

// Document ready function
$(document).ready(function() {
    if (location.href.substring(0, 47) == link) {
        setTimeout(checkRequest, 1000);
    }
})

// Check request value
let checkRequest = () => {
    let request = 0;
    request = $('#market_commodity_buyrequests').text();
    request = request.slice(25, 200);
    request = request.match(/(\d\s)?[0-9]*\,[0-9][0-9]/);
    request = request[0].toString().replace(" ", "")
    request = parseFloat(request.replace(",", "."));
    checkPrices(request);
}

// Check Market Prices
let checkPrices = (request) => {
    let i=0, price, profit;
    let profits = [];
    let results =  $('.market_listing_price_without_fee').length;

    while(i < 5 && i < results ){
        let price = $('.market_listing_price_without_fee').eq(i).text();
        price = price.match(/(\d\s)?[0-9]*\,[0-9][0-9]/);
        price = price.toString().replace(" ", "")
        price = parseFloat(price.toString().replace(",", "."));
        let profit = price - request;
        profit = profit.toFixed(2);
        profits.push(parseFloat(profit));
        i++;
    }

    show(profits , i);
};

// Show informations
let show = (profits, resoults) => {
    let i = 0;
    let color = 0;
    let text = "";

    for(i;i<resoults ;i++){
        if(profits[i] > 0)
            color = "green";
        else
            color = "red";
        text = text + "<span style='color:"+color+"; margin:10px'>"+profits[i]+"</span>";
    }

   $('<div/>',{
       html: text,
       style: "background-color:black; color:white; font-size:20px; position: fixed; top:104px; text-align:center; width: 50%; left:50%; transform:translateX(-50%);"
   }).appendTo('body');
}


TradeHelper v0.4.js
// ==UserScript==
// @require      https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js
// @name         TH 0.4
// @version      0.4
// @author       You
// @match        https://steamcommunity.com/market/listings/730/*
// @grant        none
// ==/UserScript==

// CS:GO MARKET LINK
let link = "https://steamcommunity.com/market/listings/730/";

// DOCUMENT READY - START FUNCTION
$(document).ready(function() {
    if (location.href.substring(0, 47) == link) {
        setTimeout(checkRequest, 1000);
    }
})

// CHECK REQUEST VALUE
let checkRequest = () => {
    let request = document.getElementById("market_commodity_buyrequests");
    request = request.innerHTML;
    request = request.slice(25, 200);
    request = request.match(/(\d\s)?[0-9]*\,[0-9][0-9]/);
    request = request[0].toString().replace(" ", "")
    request = parseFloat(request.replace(",", "."));
    checkPrices(request);
}

// CHECK MARKET PRICES
let checkPrices = (request) => {
    let price, profit;
    let profits = [];
    let prices = [...document.getElementsByClassName("market_listing_price_without_fee")];

    prices.forEach((element)=>{
        price = element.innerHTML;
        price = price.match(/(\d\s)?[0-9]*\,[0-9][0-9]/);
        if(price != null){
            price = price.toString().replace(" ", "");
            price = parseFloat(price.toString().replace(",", "."));
            let profit = price - request;
            profit = profit.toFixed(2);
            profits.push(parseFloat(profit));
        }
        else{
            profits.push(parseFloat(0.00));
        }
    })

  show(profits , prices.length);
};


// SHOW PROFITS
let show = (profits, resoults) => {
    let i = 0;
    let color = 0;
    let text = "";

    for(i;i<resoults ;i++){
        if(profits[i] > 0)
            color = "green";
        else
            color = "red";
        text = text + "<span style='color:"+color+"; margin:10px'>"+profits[i]+"</span></br>";
    }

   $('<div/>',{
       html: text,
       style: "background-color:black; color:white; font-size:20px; position: fixed; top:50%; transform:translateY(-50%); text-align:center; right:10px;"
   }).appendTo('body');
}